#version: '3.8'

services:
#  db:
#    image: postgres:latest
#    container_name: new_ml_postgres
#    environment:
#      POSTGRES_USER: ${DB_USER}
#      POSTGRES_PASSWORD: ${DB_PASSWORD}
#      POSTGRES_DB: ${DB_NAME}
#    ports:
#      - "5432:5432"
#    volumes:
#      - postgres_data:/var/lib/postgresql/data

  app:
    build: ./app
    container_name: python_app
    networks:
      #- bridge
      # cannot set as bridge network, because it is not a user-defined network.
      - my_custom_network
    environment:
      DB_HOST: new_ml_postgres 
      DB_PORT: ${DB_PORT}
      DB_NAME: ${DB_NAME}
      DB_USER: ${DB_USER}
      DB_PASSWORD: ${DB_PASSWORD}

networks:
  #bridge:
  my_custom_network:
    external: true

#volumes:
#  postgres_data:
